### PR TITLE
feat(droid): add Factory Droid usage adapter

### PR DESCRIPTION
**Pull request by ryoppippi**
## Try This PR

```sh
bunx -p https://pkg.pr.new/ryoppippi/ccusage@1028 ccusage droid daily --help
```

## Summary

- add a Droid adapter that reads local Factory session settings JSON files
- normalize Droid model/provider names and include thinking tokens in total usage
- wire Droid into focused commands, config schema, CLI snapshots, README, and docs

## Validation

- pnpm run format
- pnpm typecheck
- pnpm run test
- pnpm --filter docs build
- git diff --check

## Release Notes

* **New Features**
  * Added Droid as a supported coding agent CLI source with daily, monthly, and session report types
  * Includes token usage tracking and cost calculation via LiteLLM pricing, with thinking token support
  * Configurable sessions directory via `DROID_SESSIONS_DIR` environment variable

* **Documentation**
  * Updated guides to include Droid configuration and usage examples
  * Added comprehensive Droid data source documentation including setup and troubleshooting

[Review Change Stack](https://app.coderabbit.ai/change-stack/ryoppippi/ccusage/pull/1028?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: apps/ccusage/src/adapter/droid/parser.ts [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+		normalized === 'vertex' ||
+		normalized === 'vertex_ai'
+	) {
+		return 'google';
+	}
+	if (normalized === 'xai' || normalized === 'grok') {
+		return 'xai';
+	}
+	return normalized;
+}
```
[the remark this answers is not in the document]
Comment (ryoppippi) [reply apps/ccusage/src/adapter/droid/parser.ts]: @coderabbitai Fixed in the current branch. Droid provider normalization now treats x-ai / x_ai / grok as xai, and the parser regression covers providerLock: x-ai mapping to xai. Validation on the latest branch: pnpm run format, pnpm typecheck, pnpm run test, pnpm --filter docs build.

File: apps/ccusage/test/snapshots/cli-output/all-agent-daily-table.txt [hunk 2]
```diff
@@ -1,27 +1,29 @@
...
 │            │ - Codex    │          160 │        $0.00 │
 ├────────────┼────────────┼──────────────┼──────────────┤
+│            │ - Droid    │          180 │        $0.00 │
+├────────────┼────────────┼──────────────┼──────────────┤
 │            │ - OpenCode │          180 │        $0.00 │
 ├────────────┼────────────┼──────────────┼──────────────┤
 │            │ - pi-agent │          180 │        $0.00 │
 ├────────────┼────────────┼──────────────┼──────────────┤
-│ Total      │            │          880 │        $0.25 │
+│ Total      │            │        1,060 │        $0.25 │
```
[the remark this answers is not in the document]
Comment (ryoppippi) [reply apps/ccusage/test/snapshots/cli-output/all-agent-daily-table.txt]: @coderabbitai Fixed in the current branch. The all-agent compact table snapshot now uses 1,055 total tokens, matching all-agent-daily-json.txt and the direct Droid JSON total. Validation on the latest branch: pnpm run format, pnpm typecheck, pnpm run test, pnpm --filter docs build.

**Comment by ryoppippi** [comment]
@cubic-dev-ai review it

**Comment by ryoppippi** [comment]
@coderabbitai review it

**Comment by ryoppippi** [comment]
@coderabbitai review it

**Comment by ryoppippi** [comment]
@cubic-dev-ai review it

**Comment by ryoppippi** [comment]
@coderabbitai review it

**Comment by ryoppippi** [comment]
@coderabbitai review it

**Comment by ryoppippi** [comment]
@coderabbitai review

Updated this PR to merge current main and port Droid support to the Rust CLI implementation.

This now wires Droid into direct reports and all-agent reports, reads Droid *.settings.json files from DROID_SESSIONS_DIR/default ~/.factory/sessions, preserves thinking tokens in total-token accounting, and keeps Droid docs/config-schema integration.

Local validation passed:
- nix develop --command cargo test --manifest-path rust/Cargo.toml --workspace droid
- nix develop --command pnpm run format
- nix develop --command cargo test --manifest-path rust/Cargo.toml --workspace
- nix develop --command pnpm typecheck
- nix develop --command pnpm run test

**Comment by ryoppippi** [comment]
@coderabbitai review

Pushed the Droid branch after merging the latest `origin/main` and resolving conflicts against the current native Rust all-agent, Copilot/Gemini, config schema, and docs changes.

Changes in this push:
- preserved the Droid Rust adapter and wired it into current CLI/all-report/config schema code
- integrated Droid docs/navigation/environment-variable references with the current Copilot/Gemini docs
- regenerated and formatted the config schema
- removed duplicated `extra_total_tokens` initializers from the merge result

Local validation:
- `nix develop --command cargo test --manifest-path rust/Cargo.toml -p ccusage droid`
- `nix develop --command pnpm run format`
- `nix develop --command cargo test --manifest-path rust/Cargo.toml --workspace`
- `nix develop --command pnpm typecheck`
- `nix develop --command pnpm run test`

Pre-push validation also passed clippy/oxfmt/eslint hooks.

**Comment by ryoppippi** [comment]
@coderabbitai review

Merged latest main into the Droid PR, resolved conflicts with the current Rust all-agent loader/docs/schema, and pushed the update.

Local checks passed:
- pnpm --dir apps/ccusage run generate:schema
- pnpm run format
- cargo test --manifest-path rust/Cargo.toml -p ccusage droid
- cargo test --manifest-path rust/Cargo.toml --workspace
- pnpm typecheck
- pnpm run test
- pre-push oxfmt/eslint docs/clippy

**Comment by ryoppippi** [comment]
@coderabbitai review

Addressed the remaining Droid review thread and pushed a follow-up commit.

Changes:
- keep the latest Droid settings snapshot when duplicate session IDs exist
- added a regression test for archive/current duplicate snapshots

Validation:
- pnpm run format
- cargo test --manifest-path rust/Cargo.toml -p ccusage droid
- pre-push oxfmt/clippy

File: rust/crates/ccusage/src/adapter/droid.rs [hunk 3]
```diff
@@ -0,0 +1,631 @@
...
+    }
+    parsed.sort_by_key(|entry| entry.timestamp);
+    let mut seen_sessions = HashSet::new();
+    let mut entries = Vec::new();
+    for entry in parsed {
+        if !seen_sessions.insert(entry.session_id.clone()) {
+            continue;
+        }
+        entries.push(to_loaded_entry(entry, tz.as_ref(), pricing));
+    }
```
[the remark this answers is not in the document]
Comment (ryoppippi) [reply rust/crates/ccusage/src/adapter/droid.rs:145]: @coderabbitai Addressed in 863408d by keeping the newest Droid settings snapshot for duplicate session IDs and adding a regression test for archive/current duplicates. Validation: pnpm run format and cargo test --manifest-path rust/Cargo.toml -p ccusage droid.

**Comment by ryoppippi** [comment]
@coderabbitai review

Merged latest main into the Droid PR and resolved the current CLI help conflict using the namespace-only root help from main plus Droid.

Local checks passed:
- pnpm run format
- cargo test --manifest-path rust/Cargo.toml -p ccusage droid
- pre-push oxfmt/clippy

File: rust/crates/ccusage/src/output.rs [hunk 4]
```diff
@@ -76,12 +74,13 @@ pub(crate) fn totals_json(rows: &[UsageSummary]) -> Value {
...
     let cache_read = rows.iter().map(|row| row.cache_read_tokens).sum::<u64>();
+    let extra_total = rows.iter().map(|row| row.extra_total_tokens).sum::<u64>();
     let mut value = json!({
         "inputTokens": input,
         "outputTokens": output,
         "cacheCreationTokens": cache_create,
         "cacheReadTokens": cache_read,
-        "totalTokens": input + output + cache_create + cache_read,
+        "totalTokens": input + output + cache_create + cache_read + extra_total,
         "totalCost": rows.iter().map(|row| row.total_cost).sum::<f64>(),
```
[the remark this answers is not in the document]
Comment (ryoppippi) [reply rust/crates/ccusage/src/output.rs]: @coderabbitai Addressed in 9103c18 by making the non-compact table footer use totals_json.totalTokens, so extra_total_tokens are included consistently with JSON and row totals. Added totals_json_includes_extra_tokens_in_total. Validation: pnpm run format, cargo test --manifest-path rust/Cargo.toml -p ccusage droid, cargo test --manifest-path rust/Cargo.toml -p ccusage totals_json_includes_extra_tokens_in_total, and pre-push oxfmt/clippy.

**Comment by ryoppippi** [comment]
@coderabbitai review

Addressed the remaining Droid table-total review by making the non-compact footer use totals_json.totalTokens, so extra_total_tokens are included consistently with JSON and row totals.

Local checks passed:
- pnpm run format
- cargo test --manifest-path rust/Cargo.toml -p ccusage droid
- cargo test --manifest-path rust/Cargo.toml -p ccusage totals_json_includes_extra_tokens_in_total
- pre-push oxfmt/clippy

**Comment by ryoppippi** [comment]
@coderabbitai review

Synced this PR with latest main after #1031 was merged, resolved the Kimi/Droid conflicts, regenerated/formatted config schema, and pushed follow-up commits.

Local checks:
- pnpm run format
- cargo test --manifest-path rust/Cargo.toml -p ccusage droid
- cargo test --manifest-path rust/Cargo.toml -p ccusage totals_json_includes_extra_tokens_in_total
- pre-push hook: oxfmt, eslint root/docs, clippy

**Comment by ryoppippi** [comment]
@coderabbitai review

Synced this Droid PR with the latest main after #1019 landed, resolved the OpenClaw/Kimi integration conflicts, regenerated/formatted the config schema, and pushed the updates.

Local checks run:
- pnpm --dir apps/ccusage run generate:schema
- pnpm run format
- cargo test --manifest-path rust/Cargo.toml -p ccusage droid
- cargo test --manifest-path rust/Cargo.toml -p ccusage totals_json_includes_extra_tokens_in_total
- pre-push hook: oxfmt, docs eslint, clippy
